frontend/packages/console-shared/src/hooks/useTelemetry.ts

```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+import {
+  useResolvedExtensions,
+  isTelemetryListener,
+  TelemetryEventListener,
+} from '@console/dynamic-plugin-sdk';
+
+export const useTelemetry = () => {
+  const [extensions] = useResolvedExtensions(isTelemetryListener);
+  return React.useCallback<TelemetryEventListener>(
+    (eventType, properties) => {
+      extensions.forEach((e) => e.properties.listener(eventType, properties));
+    },
+    [extensions],
+  );
```

Comment:
I think we should include common properties in the `useTelemetry` hook.
For now the common properties should include the openshift version.
@spadgett anything else?

```suggestion
  const [extensions] = useResolvedExtensions(isTelemetryListener);
  const clusterVersion = useClusterVersion();
  const openshiftVersion = getOpenShiftVersion(clusterVersion);
  return React.useCallback<TelemetryEventListener>(
    (eventType, properties) => {
      extensions.forEach((e) => e.properties.listener(eventType, { openshiftVersion, ...properties }));
    },
    [extensions, openshiftVersion],
  );
```

Reply:
I'm not sure how these properties are used, but cluster ID comes to mind. Note that normal users won't be able to get the version or cluster ID. (We have a bug tracking that issue for version.) User agent also would be good if it's not already captured.

Reply:
@spadgett how will clusterID be useful and isn't that considered identifying information?
If the user cannot get the OpenShift version it should then be blank in the payload.